src/renderer/store/TimelineSpace/Modals/NewToot.js

```diff
@@ -95,15 +95,24 @@ const NewToot = {
       if (rootState.TimelineSpace.account.accessToken === undefined || rootState.TimelineSpace.account.accessToken === null) {
         throw new AuthenticationError()
       }
+      if (state.blockSubmit) {
+        return
+      }
+      commit('changeBlockSubmit', true)
       const client = new Mastodon(
         rootState.TimelineSpace.account.accessToken,
         rootState.TimelineSpace.account.baseURL + '/api/v1'
       )
       return client.post('/statuses', form)
         .then(res => {
           ipcRenderer.send('toot-action-sound')
+          commit('changeBlockSubmit', false)
           return res.data
         })
+        .catch(e => {
+          commit('changeBlockSubmit', false)
+          throw e
+        })
```

Comment:
Can you use `finally` block?

Reply:
No, NewToot.vue still has to use the error to trigger the warning message:
https://github.com/h3poteto/whalebird-desktop/blob/6c1fb2d78952f9aa5c0fb735f83cdd4319e5f952/src/renderer/components/TimelineSpace/Modals/NewToot.vue#L226-L232

Reply:
`finally` statement does not mute the error. If you use `finally` in here, the exception reach NewToot.vue.